src/test/OptionSettlement.t.sol

```diff
@@ -79,363 +83,66 @@ contract OptionSettlementTest is DSTest, NFTreceiver {
-        weth = IERC20(WETH);
-
         engine = new OptionSettlementEngine();
 
-        option = IOptionSettlementEngine.Option({
-            underlyingAsset: WETH,
-            exerciseAsset: DAI,
-            settlementSeed: 1234567,
-            underlyingAmount: 1 ether,
-            exerciseAmount: 3000 ether,
-            exerciseTimestamp: uint40(block.timestamp),
-            expiryTimestamp: (uint40(block.timestamp) + 604800)
-        });
+        IOptionSettlementEngine.Option memory option = IOptionSettlementEngine
+            .Option({
+                underlyingAsset: WETH_A,
+                exerciseAsset: DAI_A,
+                settlementSeed: 1234567,
+                underlyingAmount: 1 ether,
+                exerciseAmount: 3000 ether,
+                exerciseTimestamp: uint40(block.timestamp),
+                expiryTimestamp: (uint40(block.timestamp) + 604800)
+            });
         testOptionId = engine.newChain(option);
 
         // // Now we have 1B DAI and 1B USDC
-        writeTokenBalance(address(this), DAI, 1000000000 * 1e18);
-        writeTokenBalance(address(this), USDC, 1000000000 * 1e6);
+        writeTokenBalance(address(this), DAI_A, 1000000000 * 1e18);
+        writeTokenBalance(address(this), USDC_A, 1000000000 * 1e6);
         // // And 10 M WETH
-        writeTokenBalance(address(this), WETH, 10000000 * 1e18);
+        writeTokenBalance(address(this), WETH_A, 10000000 * 1e18);
 
-        weth.approve(address(engine), type(uint256).max);
-        dai.approve(address(engine), type(uint256).max);
+        WETH.approve(address(engine), type(uint256).max);
+        DAI.approve(address(engine), type(uint256).max);
+        USDC.approve(address(engine), type(uint256).max);
 
         // pre-load balances and approvals
         address[6] memory recipients = [
             address(engine),
-            alice,
-            bob,
-            carol,
-            dave,
-            eve
+            ALICE,
+            BOB,
+            CAROL,
+            DAVE,
+            EVE
         ];
         for (uint256 i = 0; i == 6; i++) {
             address recipient = recipients[i];
-            writeTokenBalance(recipient, DAI, 1000000000 * 1e18);
-            writeTokenBalance(recipient, USDC, 1000000000 * 1e6);
-            writeTokenBalance(recipient, WETH, 10000000 * 1e18);
+            writeTokenBalance(recipient, DAI_A, 1000000000 * 1e18);
+            writeTokenBalance(recipient, USDC_A, 1000000000 * 1e6);
+            writeTokenBalance(recipient, WETH_A, 10000000 * 1e18);
 
             VM.startPrank(recipient);
-            weth.approve(address(engine), type(uint256).max);
-            dai.approve(address(engine), type(uint256).max);
+            WETH.approve(address(engine), type(uint256).max);
+            DAI.approve(address(engine), type(uint256).max);
+            USDC.approve(address(engine), type(uint256).max);
+            // TODO(Why)
             engine.setApprovalForAll(address(this), true);
```

Comment:
This is for any future transfers that need to be done. We might as well pre-approve the contract